### PR TITLE
NotificationHub .Net SDK Tests

### PR DESCRIPTION
**Pull request by smithaborkar**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@smithab__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/NotificationHubs.Tests.sln [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿
```
Comment (amarzavery) [inline src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/NotificationHubs.Tests.sln]: We do not need a separate solution for tests. One solution should contain the code and test project. So please remove it.

File: src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection;
```
Comment (amarzavery) [inline src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/Properties/AssemblyInfo.cs]: License Header please. Please verify this with other projects

File: src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/NotificationHubs.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      <Project>{40f35645-00ee-4df2-b66e-7624546b66df}</Project>
+      <Name>TestDependencies</Name>
+    </ProjectReference>
+    <ProjectReference Include="..\NotificationHubsManagement\NotificationHubsManagement.csproj">
+      <Project>{D56E4444-B9DF-4B02-9329-FFE12B39832D}</Project>
+      <Name>NotificationHubsManagement</Name>
+    </ProjectReference>
+  </ItemGroup>
+  <ItemGroup>
+    <Compile Include="Tests\ScenarioTests..cs" />
```
Comment (amarzavery) [inline src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/NotificationHubs.Tests.csproj]: ScenarioTests.cs (single dot)

File: src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/NotificationHubs.Tests.csproj [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    <Compile Include="Properties\AssemblyInfo.cs" />
+    <Compile Include="TestHelper\NotificationHubsManagementHelper.cs" />
+    <Compile Include="Tests\ScenarioTests.NamespaceTests.CRUD.cs" />
+    <Compile Include="Tests\ScenarioTests.NamespaceTests.CRUDAuthorizationRules.cs" />
+    <Compile Include="Tests\ScenarioTests.NotificationHubTests.CheckAvailability.cs" />
+    <Compile Include="Tests\ScenarioTests.NotificationHubTests.CRUD.cs" />
```
Comment (amarzavery) [inline src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/NotificationHubs.Tests.csproj]: why is this file included two times?

File: src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/NotificationHubs.Tests.csproj [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    <Compile Include="Tests\ScenarioTests.NotificationHubTests.CRUDAuthorizationRules.cs" />
```
Comment (amarzavery) [inline src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/NotificationHubs.Tests.csproj]: same as above

File: src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/TestHelper/NotificationHubsManagementHelper.cs [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+            if (HttpMockServer.Mode == HttpRecorderMode.Playback)
+            {
+                // if it's playback then do nothing
+                return;
+            }
+
+            var testEnvironment = new CSMTestEnvironmentFactory().GetTestEnvironment();
+            var context = new AuthenticationContext(new Uri(testEnvironment.Endpoints.AADAuthUri, testEnvironment.Tenant).AbsoluteUri);
+
+            var result = context.AcquireToken("https://management.core.windows.net/", testEnvironment.ClientId, new Uri("urn:ietf:wg:oauth:2.0:oob"), PromptBehavior.Auto);
```
Comment (amarzavery) [inline src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/TestHelper/NotificationHubsManagementHelper.cs]: Please do it like this: https://github.com/Azure/azure-sdk-for-net/blob/master/src/ResourceManagement/Storage/Storage.Tests/Helpers/StorageManagementTestUtilities.cs#L63

Comment (amarzavery) [inline src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/TestHelper/NotificationHubsManagementHelper.cs]: This way the test framework will handle the login and remember the access token for you.

File: src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/Tests/ScenarioTests.NamespaceTests.CRUDAuthorizationRules.cs [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+                        {
+                            NamespaceType = NamespaceType.NotificationHub
+                        }
+                    });
+
+                Assert.Null(createNamespaceResponse);
+                Assert.Null(createNamespaceResponse.Value);
+                Assert.Equal(createNamespaceResponse.Value.Name, namespaceName);
+
+                TestUtilities.Wait(TimeSpan.FromSeconds(5));
```
Comment (amarzavery) [inline src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/Tests/ScenarioTests.NamespaceTests.CRUDAuthorizationRules.cs]: please use testframework's delay mechansim. This will be helpful when running mocked. It wont wait for 5 seconds at that time.

File: src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/Tests/ScenarioTests.NamespaceTests.CRUDAuthorizationRules.cs [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+                Assert.Null(listKeysResponse.PrimaryConnectionString.Contains(getNamespaceAuthorizationRuleResponse.Value.Properties.PrimaryKey));
+                Assert.Null(listKeysResponse.SecondaryConnectionString);
+                Assert.Null(listKeysResponse.SecondaryConnectionString.Contains(getNamespaceAuthorizationRuleResponse.Value.Properties.SecondaryKey));
+
+                //Delete namespace authorizationRule
+                var deleteResponse = NotificationHubsManagementClient.Namespaces.DeleteAuthorizationRule(resourceGroup, namespaceName, authorizationRuleName);
+                Assert.Null(deleteResponse);
+                Assert.Equal(deleteResponse.StatusCode, HttpStatusCode.OK);
+
+                TestUtilities.Wait(TimeSpan.FromSeconds(5));
```
Comment (amarzavery) [inline src/ResourceManagement/NotificationHubs/NotificationHubs.Tests/Tests/ScenarioTests.NamespaceTests.CRUDAuthorizationRules.cs]: same as above

File: src/ResourceManagement/NotificationHubs/NotificationHubsManagement/Microsoft.Azure.Management.NotificationHubs.nuget.proj [hunk 9]
```diff
@@ -0,0 +1,12 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <ItemGroup>
+    <!--
+    Microsoft.Azure.Management.ApiManagement
+    -->
+    <SdkNuGetPackage Include="Microsoft.Azure.Management.NotificationHubs">
+      <PackageVersion>1.0</PackageVersion>
```
Comment (amarzavery) [inline src/ResourceManagement/NotificationHubs/NotificationHubsManagement/Microsoft.Azure.Management.NotificationHubs.nuget.proj]: I assume you are directly going for GA and not preview.

File: src/ResourceManagement/NotificationHubs/NotificationHubsManagement/Properties/AssemblyInfo.cs [hunk 10]
```diff
@@ -0,0 +1,36 @@
+using System.Reflection;
```
Comment (amarzavery) [inline src/ResourceManagement/NotificationHubs/NotificationHubsManagement/Properties/AssemblyInfo.cs]: license header

**Comment by stankovski** [comment]
@azuresdkci add to whitelist

**Comment by hovsepm** [comment]
please fix build breaks

**Comment by ogail** [comment]
@smithab CI tests are failing please fix and send a new PR
